File: CarCareHub_/CarCareHub.Services/Database/Zaposlenik.cs
using System;
using System.Collections.Generic;

namespace CarCareHub.Services.Database; public partial class Zaposlenik
{
    public int ZaposlenikId { get; set; }
    public string? Ime { get; set; }
    public string? Prezime { get; set; }
    public int? MaticniBroj { get; set; } //dodala
    public int? BrojTelefona { get; set; } //dodala
    public Grad? Grad { get; set; }
    public int? GradId { get; set; }
    public bool Vidljivo { get; set; }

    public DateTime? DatumRodjenja { get; set; }
    public string? Email { get; set; }
    public string? Username { get; set; }
    public string? Password { get; set; }
    public string? LozinkaSalt { get; set; }
    public string? LozinkaHash { get; set; }

    public int? autoservisId { get; set; }
    public virtual Autoservis? Autoservis { get; set; }

    public int? FirmaAutodijelovaId { get; set; }
    public virtual FirmaAutodijelova? FirmaAutodijelova { get; set; }

    public int UlogaId { get; set; }
    public virtual Uloge Uloga { get; set; }

    public virtual ICollection<ChatKlijentZaposlenik> ChatKlijentZaposleniks { get; set; } = new List<ChatKlijentZaposlenik>();
    public virtual ICollection<Proizvod> Proizvods { get; set; } = new List<Proizvod>();
}

File: CarCareHub_/UI_/flutter_mobile/lib/models/zaposlenik.dart
import 'package:flutter_mobile/models/autoservis.dart';

import 'package:flutter_mobile/models/firmaautodijelova.dart';
import 'package:flutter_mobile/models/grad.dart';
import 'package:flutter_mobile/models/uloge.dart';
import 'package:json_annotation/json_annotation.dart';

part 'zaposlenik.g.dart';

@JsonSerializable()
class Zaposlenik {
  int? zaposlenikId;
  String? ime;
  String? prezime;
  int? maticniBroj;
  int? brojTelefona;
  final Grad? grad;
  final int? gradId;
  DateTime? datumRodjenja;
  String? email;
  String? username;
  String? lozinkaSalt;
  String? lozinkaHash;
  String? password;
  int? ulogaId;
  final Uloge? uloga;
  int? autoservisId;
  final Autoservis? autoservis;
  int? firmaAutodijelovaId;
  final FirmaAutodijelova? firmaAutodijelova;

  String? passwordAgain;

  Zaposlenik({
    this.zaposlenikId,
    this.ime,
    this.prezime,
    this.maticniBroj,
    this.brojTelefona,
    this.gradId,
    this.datumRodjenja,
    this.email,
    this.username,
    this.lozinkaSalt,
    this.lozinkaHash,
    this.password,
    this.ulogaId,
    this.autoservisId,
    this.firmaAutodijelovaId,
    this.grad,
    this.autoservis,
    this.firmaAutodijelova,
    this.uloga,
    
  });

  factory Zaposlenik.fromJson(Map<String, dynamic> json) =>
      _$ZaposlenikFromJson(json);

  Map<String, dynamic> toJson() => _$ZaposlenikToJson(this);
}
